src/content/en/fundamentals/performance/webpack/use-long-term-caching.md

```diff
@@ -0,0 +1,647 @@
+  docs (for Vue.js)
+
+### For traditional multi-page apps
+
+To split traditional apps by pages, use webpack’s [entry
+points](https://webpack.js.org/concepts/entry-points/). If your app has three kinds of pages: the
+home page, the article page and the user account page, – it should have three entries:
+
+    // webpack.config.js
+    module.exports = {
+      entry: {
+        home: './src/Home/index.js',
+        article: './src/Article/index.js',
+        profile: './src/Profile/index.js'
+      },
+    };
+
+For each entry file, webpack will build a separate dependency tree and generate a bundle that
+includes only modules that are used by that entry:
+
+<pre class="prettyprint">
+$ webpack
+Hash: 318d7b8490a7382bf23b
+Version: webpack 3.8.1
+Time: 4273ms
+<strong>                            Asset     Size  Chunks             Chunk Names</strong>
+      ./0.8ecaf182f5c85b7a8199.js  22.5 kB       0  [emitted]
+   ./home.91b9ed27366fe7e33d6a.js    18 kB       1  [emitted]  home
+./article.87a128755b16ac3294fd.js    32 kB       2  [emitted]  article
+./profile.de945dc02685f6166781.js    24 kB       3  [emitted]  profile
+ ./vendor.4f14b6326a80f4752a98.js    46 kB       4  [emitted]  vendor
+./runtime.318d7b8490a7382bf23b.js  1.45 kB       5  [emitted]  runtime
+</pre>
+
+So, if only the article page uses Lodash, the `home` and the `profile` bundles won’t include it –
+and the user won’t have to download this library when visiting the home page.
+
+Separate dependency trees have their drawbacks though. If two entry points are using Lodash, both
+will include a copy of it. To solve this, use the
+[`CommonsChunkPlugin`](https://webpack.js.org/plugins/commons-chunk-plugin/) – it will move common
+dependencies into a separate file:
+
+    // webpack.config.js
+    module.exports = {
+      plugins: [
+        new webpack.optimize.CommonsChunkPlugin({
+          // A name of the chunk that will include the common dependencies
+          name: 'common',
+
+          // The plugin will move a module into a common file
+          // only if it’s included into `minChunks` chunks
+          // (Note that the plugin analyzes all chunks, not only entries)
+          minChunks: 2,    // 2 is the default value
+        }),
+      ],
+    };
+
+Feel free to play with the `minChunks` value to find the best one. Generally, you want to keep it
+small, but increase if the number of chunks grows. For example, for 3 chunks, `minChunks` might be
+2, but for 30 chunks, it might be 8 – because if you keep it at 2, too many modules will get into
+the common file, inflating it too much.
+
+### Further reading {: .hide-from-toc }
+
+* Webpack docs [about the concept of entry points](https://webpack.js.org/concepts/entry-points/)
+
+* Webpack docs [about the CommonsChunkPlugin](https://webpack.js.org/plugins/commons-chunk-plugin/)
+
+* [“Getting the most out of the CommonsChunkPlugin”](https://medium.com/webpack/webpack-bits-
```

Comment:
don't linebreak URLs, otherwise the link breaks. it's OK if this exceeds 80 chars. 

if you really wanted to line break this, you could do something like:

```
* [“Getting the most out of the
CommonsChunkPlugin”](https://medium.com/webpack/webpack-bits-more-and-more-and-more-and-more)
```

Reply:
> don't linebreak URLs, otherwise the link breaks

OK! (I checked locally, and they worked, but I’ll fix them.) Will fix this shortly.

> if you really wanted to line break this, you could do something like:

I’ve actually got test fails because of this: https://travis-ci.org/google/WebFundamentals/builds/325794409#L743. (If you see [the line which triggers the error](https://github.com/google/WebFundamentals/commit/3c3ac29abc1ce4380e9fcafabf9e325dcd85ed8f#diff-9f55b5ed9c2b6c50e93df9647ab49499R158), there’s a line break inside the link name.)